### PR TITLE
Don't review a file if its signing failed (bug 1156333)

### PR DESCRIPTION
**Pull request by magopian**
Fixes [bug 1156333](https://bugzilla.mozilla.org/show_bug.cgi?id=1156333)

This is more or less a revamp of #524

### CODE REVIEW

**Comment by robhudson** [comment]
I'm not seeing where the juice of this commit is. All the non-test changes look like just changing how you're passing the signing server setting?

File: lib/crypto/packaged.py [hunk 1]
```diff
@@ -45,13 +45,9 @@ def supports_firefox(file_obj):
...
-def get_endpoint(file_obj):
-    """Get the endpoint to sign the file, depending on its review status."""
-    server = settings.SIGNING_SERVER
-    if (file_obj.status == amo.STATUS_BETA or
-            file_obj.version.addon.status != amo.STATUS_PUBLIC):
-        server = settings.PRELIMINARY_SIGNING_SERVER
-    if not server:
+def get_endpoint(server):
+    """Get the endpoint to sign the file, either the full or prelim one."""
+    if not server:  # Setting is empty, signing isn't enabled.
```
Comment (robhudson) [inline lib/crypto/packaged.py]: is this the meat of it? I guess I was expecting a `try`/`except` block to catch signing errors?

**Comment by magopian** [comment]
the way the signing server is passed is just a "side effect" of the change. The meat of it is doing the signing the first thing in the review helpers. By doing the signing first, if it fails, the process stops there, and we don't end in a weird state with addons that are reviewed but not signed.

What we want is that the addons stay in the same state... until they're signed, and then we can update their status.

Passing the signing server explicitely allows us to do the signing before the addon/file has the final status (which was previously used to infer which signing server to use).

**Comment by robhudson** [comment]
Ah ok, so you're not catching the exception but letting it bubble up. Got it. I was looking for exception catching and stuff. This looks good to me then. r+

**Pull request merged**